Shared/Views/NewSubView.swift

```diff
@@ -129,71 +138,62 @@ fileprivate struct OptionsCell: View {
     }
 }
 
-
 fileprivate struct CyclePickerCell: View {
+    @EnvironmentObject var partialSheetManager: PartialSheetManager
+
     @State var title: AppLocal.Strings
-//    @Binding var value: String
+    @Binding var value: Cycle
```

Comment:
Here you should use a protocol CyclePickerContentDisplay protocol, where Cycle and Duration implement this protocol. So same View can easily be used for both input fields.

Reply:
ya we could have done it with protocol, check the current handlation ..